### PR TITLE
Add TypeScript definitions.

### PR DESCRIPTION
**Pull request by fimars**
Ref https://github.com/DefinitelyTyped/DefinitelyTyped/issues/27519
And inspire #123 (Don't know why this PR closed.

CC @edmorley

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+	interface PluginClass {
+		new (...opts: any[]): webpack.Plugin;
+	}
+
+	interface Orderable {
+		before(name: string): this;
+		after(name: string): this;
+	}
+}
```
Comment (edmorley) [inline types/index.d.ts]: Could you add the missing newline at the end of this file (and the others too)?

Comment (fimars) [reply types/index.d.ts]: should

**Comment by yordis** [comment]
@edmorley would you take a PR using TypeScript for this project? 

We do not need to turn strict rules on so it will feel like JavaScript so we do not need to fight the tool but we can take advantage of it.

If you will maintain this file at the end, I see more value on using TypeScript.

I take the take the responsibility to convert the project so you can focus on something else.

**Comment by yordis** [comment]
💔 <--- On reactions 

Any particular reason? You can turn all the TypeScript noise off and maintain the typespec as you are doing it today but in the code 💔

**Comment by eliperelman** [comment]
@yordis we don't use TypeScript, and maintaining tooling and syntax around a language we don't use increases our maintenance burden.

File: types/test/webpack-chain-tests.ts [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+      .add('.js')
+      .end()
+    .packageMains
+      .add('index.js')
+      .end()
+    .end()
+
+  .merge({})
+  .toConfig();
+
```
Comment (eliperelman) [inline types/test/webpack-chain-tests.ts]: Remove extra blank line from end of file.

Comment (fimars) [reply types/test/webpack-chain-tests.ts]: ok

**Comment by edmorley** [comment]
Sorry for the delay here. Could you rebase on master and adjust the typings to account for #133?

@eliperelman once those changes made, any other items you'd like to see addressed, or good to merge?

**Comment by fimars** [comment]
> Sorry for the delay here. Could you rebase on master and adjust the typings to account for #133?
> 
> @eliperelman once those changes made, any other items you'd like to see addressed, or good to merge?

I adjusted the type declaration file against the latest documentation. review need.
@edmorley @eliperelman

**Comment by edmorley** [comment]
Released in webpack-chain v5.2.0:
https://www.npmjs.com/package/webpack-chain/v/5.2.0

**Pull request merged**